### PR TITLE
feat: add icon to shutdown on AppBar

### PR DESCRIPTION
**Pull request by BastelKlug**
Closes: #729 

- Shutdown icon on AppBar/TopBar
- Entry in the settings to (de)activate

Here you can see it in action:

https://user-images.githubusercontent.com/44060099/177038146-131b4d90-e044-4868-8f4e-533ad995fa6f.mov

### CODE REVIEW

**Review by matmen — DISMISSED** [review]
Thanks for your PR! This looks good to me already, though I'm going to mention my UX concerns here again (https://github.com/fluidd-core/fluidd/pull/726#issuecomment-1162354264): Moving the e-stop button conditionally may break its UX (muscle memory wise), though I think it's not too bad in this case (considering the user actively has to toggle it and it doesn't toggle visibility based on an internal state). In the above mentioned PR the button was moved to the left of the e-stop button, but I'm fine with either position in this case, so fully up to you.

**Review by pedrolamas — CHANGES_REQUESTED** [review]
I wonder if we should be more specific and instead of just "Shutdown" have "Host Shutdown" as that is what it is all about... @matmen what do you think?

File: src/components/layout/AppBar.vue [hunk 1]
```diff
@@ -199,6 +226,19 @@ export default class AppBar extends Mixins(StateMixin) {
...
+      this.$tc('app.general.simple_form.msg.confirm_shutdown_host'),
+      { title: this.$tc('app.general.label.confirm'), color: 'card-heading', icon: '$error' }
+    )
+      .then(res => {
+        if (res) {
+          this.$emit('click')
+          this.hostShutdown()
+        }
+      })
+  }
```
Comment (pedrolamas) [inline src/components/layout/AppBar.vue:240]:
```suggestion
  async handleHostShutdown () {
    const res = await this.$confirm(
      this.$tc('app.general.simple_form.msg.confirm_shutdown_host'),
      { title: this.$tc('app.general.label.confirm'), color: 'card-heading', icon: '$error' }
    )

    if (res) {
      this.hostShutdown()
    }
  }
```

That `$emit` call should not be needed here (that was on the `SystemCommands.vue` as it probably was needed to dismiss the popup menu, but that is not the case here!

Ideally, I think we should move from `Promise` based code to more modern `async/await` calls, so I went ahead and refactored this method!

**Comment by matmen** [comment]
> I wonder if we should be more specific and instead of just "Shutdown" have "Host Shutdown" as that is what it is all about...

@pedrolamas Sure, sounds good to me

**Comment by BastelKlug** [comment]
Thank you for your improvements. 

I have updated the code, so the text looks like this:
Bildschirmfoto 2022-07-03 um 23 12 00

**Review by pedrolamas — APPROVED** [review]

**Comment by Martyfree123** [comment]
Thank you to @BastelKlug for making my feature request into a reality. This is awesome.

**Comment by Martyfree123** [comment]
When will this be included in a stable release?

**Comment by Martyfree123** [comment]
Awesome thank you!

**Comment by matmen** [comment]
Actually @Martyfree123, I just realized your feature request asked for a button to toggle _printer_ power.. This PR added a button to shut down the _host_ (Raspberry Pi), so I think there was some kind of misinterpretation of your FR.. Just to clarify - would you like a toggle button for the `Power` side-bar entries (image)?

**Comment by Martyfree123** [comment]
Yes, I was looking for a button to toggle printer power instead of digging into the side bar for it.

**Comment by Martyfree123** [comment]
I missed that from the video as well. Damn I was really excited :/

**Pull request merged**